public/celestia-app.sh

```diff
@@ -0,0 +1,140 @@
+LOGFILE="$HOME/celestia-temp/logfile.log"
+TEMP_DIR="$HOME/celestia-temp"
+
+# Check if the directory exists
+if [ -d "$TEMP_DIR" ]; then
+    read -p "Directory $TEMP_DIR exists. Do you want to clear it out? (y/n) " -n 1 -r
+    echo    # move to a new line
+    if [[ $REPLY =~ ^[Yy]$ ]]
+    then
+        rm -rf "$TEMP_DIR"
+        echo "Directory $TEMP_DIR has been removed."
+    fi
+fi
+
+mkdir -p "$TEMP_DIR"
+touch "$LOGFILE"
+
+# Log and print the log file location
+echo "Log file is located at: $LOGFILE" | tee -a "$LOGFILE"
+
+# Create a temporary directory to work from
+TEMP_DIR="$HOME/celestia-temp"
+mkdir -p "$TEMP_DIR"
+cd "$TEMP_DIR"
+
+# Log and print a message
+echo "Working from temporary directory: $TEMP_DIR" | tee -a "$LOGFILE"
+
+# Comment out once all releases haev prebuilt binaries
+# Set the version manually, until all releases have prebuilt
+# binaries attached to them
+VERSION=v1.1.0
+
+# # Uncomment this once all releases have prebuilt binaries
+# # Fetch the latest release tag from GitHub
+# VERSION=$(curl -s "https://api.github.com/repos/celestiaorg/celestia-app/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+
+# Log and print a message
+echo "Latest version detected: $VERSION" | tee -a "$LOGFILE"
+
+# Detect the operating system and architecture
+OS=$(uname -s)
+ARCH=$(uname -m)
+
+# Translate architecture to expected format
+case $ARCH in
+    x86_64)
+        ARCH="x86_64"
+        ;;
+    aarch64|arm64)
+        ARCH="arm64"
+        ;;
+    *)
+        echo "Unsupported architecture." | tee -a "$LOGFILE"
+        exit 1
+        ;;
+esac
+
+# Translate OS to expected format
+case $OS in
+    Linux|Darwin)
+        ;;
+    *)
+        echo "Unsupported operating system." | tee -a "$LOGFILE"
+        exit 1
+        ;;
+esac
+
+# Construct the download URL
+PLATFORM="${OS}_${ARCH}"
+URL="https://github.com/celestiaorg/celestia-app/releases/download/$VERSION/celestia-app_$PLATFORM.tar.gz"
+
+# Log and print a message
+echo "Downloading from: $URL" | tee -a "$LOGFILE"
+
+# Download the tarball
+if ! wget "$URL" >> "$LOGFILE" 2>&1; then
+    echo "Download failed. Exiting." | tee -a "$LOGFILE"
+    exit 1
+fi
+
+# Detect if running on macOS and use appropriate command for checksum
+if [ "$OS" = "Darwin" ]; then
+    CALCULATED_CHECKSUM=$(shasum -a 256 "celestia-app_$PLATFORM.tar.gz" | awk '{print $1}')
+else
+    CALCULATED_CHECKSUM=$(sha256sum "celestia-app_$PLATFORM.tar.gz" | awk '{print $1}')
+fi
+
+# Download checksums.txt
+if ! wget "https://github.com/celestiaorg/celestia-app/releases/download/$VERSION/checksums.txt" >> "$LOGFILE" 2>&1; then
+    echo "Failed to download checksums. Exiting." | tee -a "$LOGFILE"
+    exit 1
+fi
+
+# Find the expected checksum in checksums.txt
+EXPECTED_CHECKSUM=$(grep "celestia-app_$PLATFORM.tar.gz" checksums.txt | awk '{print $1}')
+
+# Verify the checksum
+if [ "$CALCULATED_CHECKSUM" != "$EXPECTED_CHECKSUM" ]; then
```

Comment:
is this a sufficient checksum verification @MSevey?

Reply:
IMO sufficient and the steps here are even a bit better than https://github.com/celestiaorg/celestia-app/blob/b786b081b18b8eca9af45bd453c71012bb941d3f/README.md?plain=1#L73-L83 b/c I didn't know that on Linux machines one needs to use `sha256sum` instead of `shasum -a 256` on Apple machines

Reply:
only thing I would check is that `CALCULATED_CHECKSUM` is present. I'm not sure how scope is managed in bash, but if we only define that variable in the previous if statement, it might not be accessible. In which case you would just want to define it prior to calculating it.
```
CALCULATED_CHECKSUM=''
```

Reply:
yep, it's still accessible. it looks like the variable is only [local if you define it that way](https://tldp.org/LDP/abs/html/localvar.html). and for blocks of code: "Block of code [curly brackets]. Also referred to as an inline group, this construct, in effect, creates an anonymous function (a function without a name). However, unlike in a "standard" [function](https://tldp.org/LDP/abs/html/functions.html#FUNCTIONREF), the variables inside a code block remain visible to the remainder of the script." ([link](https://tldp.org/LDP/abs/html/special-chars.html#CODEBLOCKREF:~:text=Block%20of%20code%20%5Bcurly%20brackets%5D.%20Also%20referred%20to%20as%20an%20inline%20group%2C%20this%20construct%2C%20in%20effect%2C%20creates%20an%20anonymous%20function%20(a%20function%20without%20a%20name).%20However%2C%20unlike%20in%20a%20%22standard%22%20function%2C%20the%20variables%20inside%20a%20code%20block%20remain%20visible%20to%20the%20remainder%20of%20the%20script.))